error.go

```diff
@@ -338,11 +338,22 @@ const (
 	SQLITE_AUTH_USER               = ErrorCode(C.SQLITE_AUTH_USER)
 )
 
+type causer interface {
+	Cause() error
+}
+
 // ErrCode extracts the SQLite error code from err.
 // If err is not a sqlite Error, SQLITE_ERROR is returned.
 // If err is nil, SQLITE_OK is returned.
+//
+// This function supports wrapped errors that implement
+// `Cause() error`, like https://github.com/pkg/errors generates.
```

Comment:
Go doc comments aren't markdown. Use a tab to get godoc to recognize code:

// This function will unwrap errors that implement
//\tinterface{ Cause() error }
// for errors from packages like github.com/pkg/errors.